src/runtime_src/tools/scripts/loader

```diff
@@ -48,6 +48,11 @@ fi
 
 . "${XRT_SETUP_SCRIPT}" > /dev/null
 
+# XRT uses system librairies whereas Vitis and Vivado ship there own.
+# when XRT tools are invoked from Vitis/Vivado the LD_LIBRARY_PATH will be set to use libraries shipped with Vitis/vivado
+# so we reset the LD_LIBRARY_PATH to make XRT tool select the correct libraries.
+LD_LIBRARY_PATH=$XILINX_XRT/lib
+
```

Comment:
Making progress.
But I would prefer you make more precise your previous LD_LIBRARY_PATH patching solution and you move it here.

Reply:
I think you still need to remove offensive Vitis content from `$LIBRARY_PATH` as you did first. Just need a filtering targeting better Vivado & Vitis.